config/gnu-warnings/error-general

```diff
@@ -3,18 +3,22 @@
 # circumstances, so ask the compiler to treat them as errors:
 #
 -Werror=bad-function-cast
+-Werror=cast-align
+-Werror=format
 -Werror=implicit-function-declaration
+-Werror=int-to-pointer-cast
 -Werror=missing-declarations
 -Werror=missing-prototypes
 -Werror=nested-externs
 -Werror=old-style-definition
 -Werror=packed
 -Werror=pointer-sign
 -Werror=pointer-to-int-cast
--Werror=int-to-pointer-cast
 -Werror=redundant-decls
 -Werror=strict-prototypes
 -Werror=switch
+-Werror=unused-but-set-variable
+-Werror=unused-variable
```

Comment:
For example, this is a silly -Werror to turn on in the general case. Failing to build because of an unused variable is ridiculous. Perfectly ok in a developer build mode because it encourages us to have cleaner code, but even then developers may end up needing to disable it due to an external dependency being built.